modules/tensor_mechanics/src/materials/ComputeMeanThermalExpansionEigenstrainBase.C

```diff
@@ -28,12 +28,14 @@ ComputeMeanThermalExpansionEigenstrainBase::ComputeMeanThermalExpansionEigenstra
 void
 ComputeMeanThermalExpansionEigenstrainBase::initialSetup()
 {
-  _alphabar_stress_free_temperature = meanThermalExpansionCoefficient(_stress_free_temperature);
+  _alphabar_stress_free_temperature =
+      meanThermalExpansionCoefficient(_stress_free_temperature[_qp]);
   _thexp_stress_free_temperature =
-      _alphabar_stress_free_temperature * (_stress_free_temperature - referenceTemperature());
+      _alphabar_stress_free_temperature * (_stress_free_temperature[_qp] - referenceTemperature());
 
-  // Evaluate the derivative so it will error out early on if there are any issues with that
-  meanThermalExpansionCoefficientDerivative(_stress_free_temperature);
+  // Evaluate the derivative so it will error out early on if there are any
+  // issues with that
+  meanThermalExpansionCoefficientDerivative(_stress_free_temperature[_qp]);
```

Comment:
This needs to be removed now. `_qp` is not set (and the variables not prepared) in `initialSetup` and we might just as well error out later.